File: E_Kotlin/Projects/AndroidStudio/XML/PoliceDriverGame/app/src/main/java/com/sebss/juegosnake/androidimpl/AndroidFileIO.java
package com.sebss.juegosnake.androidimpl;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.nio.file.Files;
import java.nio.file.Paths;

import android.content.res.AssetManager;
import android.os.Environment;
import com.sebss.juegosnake.FileIO;

public class AndroidFileIO implements FileIO {
    AssetManager assets;
    String rutaAlmacenamientoExterno;

    public AndroidFileIO(AssetManager assets) {
        this.assets = assets;
        this.rutaAlmacenamientoExterno = Environment
                .getExternalStorageDirectory().getAbsolutePath()
                + File.separator;
    }

    @Override
    public InputStream leerAsset(String nombreArchivo) throws IOException {
        return assets.open(nombreArchivo);
    }

    @Override
    public InputStream leerArchivo(String nombreArchivo) throws IOException {
        return Files.newInputStream(Paths.get(rutaAlmacenamientoExterno + nombreArchivo));
    }

    @Override
    public OutputStream escribirArchivo(String nombreArchivo) throws IOException {
        return Files.newOutputStream(Paths.get(rutaAlmacenamientoExterno + nombreArchivo));
    }

}

File: E_Kotlin/Projects/template_games/androidimpl/AndroidFileIO.kt
package com.sebss.pass_word.androidimpl

import android.content.res.AssetManager
import android.os.Environment
import com.sebss.pass_word.interfaces.FileIO
import java.io.File
import java.io.IOException
import java.io.InputStream
import java.io.OutputStream
import java.nio.file.Files
import java.nio.file.Paths

class AndroidFileIO(var assets: AssetManager) : FileIO {
    private var externalStoragePath: String = (Environment
        .getExternalStorageDirectory().absolutePath
            + File.separator)

    @Throws(IOException::class)
    override fun readAsset(fileName: String): InputStream {
        return assets.open(fileName)
    }

    @Throws(IOException::class)
    override fun readFile(fileName: String): InputStream {
        return Files.newInputStream(Paths.get(externalStoragePath + fileName))
    }

    @Throws(IOException::class)
    override fun writeFile(fileName: String): OutputStream {
        return Files.newOutputStream(Paths.get(externalStoragePath + fileName))
    }
}